tools/bump-pack/src/pack.ts

```diff
@@ -22,6 +23,40 @@ export async function pack(options: PackOptions) {
   if (dryRun) {
     console.log(`Would have run "npm pack" in ${packageDir}`);
   } else {
-    execSync("npm pack", { cwd: packageDir, stdio: "inherit" });
+    preparePackageJsonAndRun(packageDir, () => {
+      execSync("npm pack", { cwd: packageDir, stdio: "inherit" });
+    });
+  }
+}
+
+/**
+ * Transforms the package.json at `packageDir` to use the publishConfig field and remove other unnecessary fields such as devDependencies.
+ * @param packageDir The directory that contains the package.json to transform.
+ * @param callback The function called after the transformation, before reverting the package.json back to its original state.
+ */
+function preparePackageJsonAndRun(packageDir: string, callback: () => void) {
+  const packageJsonPath = `${packageDir}/package.json`;
+  const packageJson = JSON.parse(readFileSync(packageJsonPath, "utf8"));
```

Comment:
Maybe save the text instead, then save that to the file later. Avoids unnecessary processing and any possible changes to the original file.

```suggestion
	const packageJsonPath = `${packageDir}/package.json`;
  const packageJsonString = readFileSync(packageJsonPath, "utf8");
  const packageJson = JSON.parse(packageJsonString);
```

Reply:
That's what I had initially 👍🏻 Makes sense.